packages/kit/src/runtime/server/fetch.js

```diff
@@ -51,15 +51,19 @@ export function create_fetch({ event, options, manifest, state, get_cookie_heade
 				}
 
 				if (url.origin !== event.url.origin) {
-					// allow cookie passthrough for "same-origin"
+					// credentials: 'include' will always include cookies
+					// credentials: 'same-origin' depends on the origin
```

Comment:
```suggestion
					// Do not forward anything for "credentials: same-origin" because we don't know
					// which cookie belongs to which domain (browser does not pass this info) so
					// it's not safe to foward any of them.
					// Allow cookie passthrough for "credentials: same-origin"
```